Source/TrackableEntities.EF.5/DbContextExtensions.cs

```diff
@@ -199,16 +199,31 @@ public static void ApplyChanges(this InterceptorPool pool, IEnumerable<ITrackabl
                     context.ApplyChangesOnProperties(item, visitationHelper.Clone(), TrackingState.Deleted, interceptors);
                 }
 
+                var isComplex = context.IsComplexType(item.GetType());
+                bool isParentEligible(TrackingState parentState) => parentState == TrackingState.Modified || parentState == TrackingState.Unchanged;
+                var isModifiable = 
+                          (isComplex 
+                            ? isParentEligible(parent.TrackingState) 
+                            : (item.TrackingState == TrackingState.Modified && (state == null || state == TrackingState.Modified)))                          
+                          && item.ModifiedProperties?.Count > 0;
+
                 // Set modified properties
-                if (item.TrackingState == TrackingState.Modified
-                    && (state == null || state == TrackingState.Modified)
-                    && item.ModifiedProperties != null
-                    && item.ModifiedProperties.Count > 0)
+                if (isModifiable)
```

Comment:
I would suggest creating an `IsModifiable(ITrackable item)`  method to encapsulate all the logic.

```csharp
private bool IsModifiable(ITrackable item) {
    if (item.TrackingState == TrackingState.Modified
        && (state == null || state == TrackingState.Modified)
        && item.ModifiedProperties != null
        && item.ModifiedProperties.Count > 0) {
            if (!context.IsComplexType(item.GetType())) return true;
        return parent.TrackingState == TrackingState.Modified || parent.TrackingState == TrackingState.Unchanged;
    }
    return false;
}
```

Check the syntax, but I think this will do the trick.